src/etc/schema.graphqls

```diff
@@ -0,0 +1,39 @@
+# ScandiPWA - Progressive Web App for Magento
+#
+# Copyright © Scandiweb, Inc. All rights reserved.
+# See LICENSE for license details.
+#
+# @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+# @package scandipwa/compare-graphql
+# @link https://github.com/scandipwa/compare-graphql
+#
+# Copyright © Magento, Inc. All rights reserved.
+# See COPYING.txt for license details.
+
+
+type Mutation {
+    addProductToCompare(product_id: Int!, guestCartId: String): Boolean @resolver(class: "\\ScandiPWA\\CompareGraphQl\\Model\\Resolver\\AddProductToCompare")
+    removeComparedProduct(product_id: Int!, guestCartId: String): Boolean @resolver(class:"\\ScandiPWA\\CompareGraphQl\\Model\\Resolver\\RemoveCompareProduct")
+    clearComparedProducts(guestCartId: String): Boolean @resolver(class:"\\ScandiPWA\\CompareGraphQl\\Model\\Resolver\\ClearCompareProducts")
+}
+
+type Query {
+    compareProducts(guestCartId: String): CompareProductsOutput @resolver(class: "\\ScandiPWA\\CompareGraphQl\\Model\\Resolver\\CompareProductsResolver") @doc(description: "The compare products query returns the contents of a customer's compared products")
+}
+
+type CompareProductsOutput {
+    products: [ProductInterface] @doc(description: "Compare list items")
+    count: Int @doc(description: "The number of items in the compare list")
+}
+
+extend interface ProductInterface {
+    comparableAttributes: [AttributeWithValue] @resolver(class: "\\ScandiPWA\\CompareGraphQl\\Model\\Resolver\\ComparableAttributesResolver")
+}
+
+type AttributeWithValue {
+    attribute_id: Int
+    attribute_label: String
+    attribute_code: String
+    attribute_value: String
+    attribute_type: String
+}
```

Comment:
I think implementing comments above we can survive with old arguments.

Reply:
Fixed.